### PR TITLE
wip: add SMART disk monitor collector

### LINKED ISSUE #78: Feature Request: SMART-values
the SMART values should pr provided by WMI according to 
https://exchange.nagios.org/directory/Plugins/Operating-Systems/Windows/NRPE/check_smartwmi-SMART-Monitoring-for-Windows-by-using-builtin-WMI/details

that would be great!

### PR DESCRIPTION
**Pull request by martinlindhe**
First part to resolve #78 

In order to continue, we need to get https://github.com/StackExchange/wmi/pull/30 accepted,
because the SMART data is in a byte array and StackExchange/wmi currently doesnt support []uint8.

Next issue is that this requires to be ran as admin, or the query will return empty result.
You can check manually what data you get by issuing `gwmi -namespace root\wmi -class MSStorageDriver_ATAPISmartData`

Third, while developing the class stopped returning values multiple times, which I had to resolve by rebooting the box. I hope that was only due to my negligence. Will keep it running for a while to find out.

The current data exported looks like this:
```

# HELP wmi_smart_capability Smart capability (SMART.SmartCapability)
# TYPE wmi_smart_capability gauge
wmi_smart_capability{volume="SCSI\\Disk&Ven_Corsair&Prod_Performance_Pro\\4&26e76114&0&050000_0"} 2
wmi_smart_capability{volume="SCSI\\Disk&Ven_WDC&Prod_WD10EACS-00D6B1\\4&26e76114&0&030000_0"} 3
wmi_smart_capability{volume="SCSI\\Disk&Ven_WDC&Prod_WD20EARS-00MVWB0\\4&26e76114&0&040000_0"} 3
# HELP wmi_smart_selftest_status The self test status code (SMART.SelfTestStatus)
# TYPE wmi_smart_selftest_status gauge
wmi_smart_selftest_status{volume="SCSI\\Disk&Ven_Corsair&Prod_Performance_Pro\\4&26e76114&0&050000_0"} 0
wmi_smart_selftest_status{volume="SCSI\\Disk&Ven_WDC&Prod_WD10EACS-00D6B1\\4&26e76114&0&030000_0"} 0
wmi_smart_selftest_status{volume="SCSI\\Disk&Ven_WDC&Prod_WD20EARS-00MVWB0\\4&26e76114&0&040000_0"} 0
# HELP wmi_smart_total_time Total time used (SMART.TotalTime)
# TYPE wmi_smart_total_time gauge
wmi_smart_total_time{volume="SCSI\\Disk&Ven_Corsair&Prod_Performance_Pro\\4&26e76114&0&050000_0"} 10
wmi_smart_total_time{volume="SCSI\\Disk&Ven_WDC&Prod_WD10EACS-00D6B1\\4&26e76114&0&030000_0"} 23400
wmi_smart_total_time{volume="SCSI\\Disk&Ven_WDC&Prod_WD20EARS-00MVWB0\\4&26e76114&0&040000_0"} 36960
```

### CODE REVIEW

**Comment by martinlindhe** [comment]
After running a while, now only 1 of 3 disks which was initially returned from the query is returned.
I believe it is caused by some powersaving feature...

File: docs/SMART.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+To see what disks has S.M.A.R.T. available on your machine, you can run
+    
+    gwmi -namespace root\wmi -class MSStorageDriver_ATAPISmartData
+
+
+### Disappearing metrics
+
+If you experience gaps in the collected smart metrics, most likely Windows put the hard disk to sleep.
+
+To prevent the Hard Disk from going to sleep, click on the Battery / Power icon in the taskbar and select More Power options. In the Control Panel windows which opens, select Change Plan settings for your current Power Plan. In the next window, select Change advanced power settings.
```
Comment (carlpett) [inline docs/SMART.md]: We might want to have a note about this maybe not being something to be done lightly? Afaik, not spinning down the disks will both reduce lifetime as well as increase power consumption.

**Comment by carlpett** [comment]
This looks like it will be a nice addition, great work! What is the state of it?

**Comment by martinlindhe** [comment]
Some work remains, basically I've ported over the smart data parser from this old and somewhat buggy nagios script. The code needs a bunch of restructuring and testing.

**Comment by martinlindhe** [comment]
Closed due to no progress. Also, https://github.com/StackExchange/wmi/pull/30 has recieved no input

**Comment by r4j4h** [comment]
🎉 StackExchange/wmi#30 has been merged!

**Pull request closed without merging**